src/pixelator/graph/utils.py

```diff
@@ -330,3 +366,94 @@ def update_edgelist_membership(
 
     logger.debug("Membership in edge list updated")
     return edgelist
+
+
+def _update_edgelist_membership_lazyframe(
+    edgelist: pl.LazyFrame,
+    graph: Optional[Graph] = None,
+    prefix: Optional[str] = None,
+) -> pl.LazyFrame:
+    if prefix is None:
+        prefix = DEFAULT_COMPONENT_PREFIX
+
+    if "component" in edgelist.columns:
+        logger.info("The input edge list already contain a component column")
+
+    if not graph:
+        graph = Graph.from_edgelist(
+            edgelist=edgelist,
+            add_marker_counts=False,
+            simplify=False,
+            use_full_bipartite=True,
+        )
+
+    logger.debug("Searching for connected components")
+    connected_components = graph.connected_components()
+
+    logger.debug("Building edge to component mappings")
+    edge_index_to_component_mapping = {
+        e.index: component_idx
+        for component_idx, component in enumerate(connected_components)
+        for e in graph.es.select_within({v.index for v in component.vertices()})
+    }
+
+    logger.debug("Mapping components on the edge list")
+    edgelist_with_component_info = (
```

Comment:
Jesus! 😆

Reply:
You are going to need to explain this to me tomorrow.

Reply:
Haha! Yeah. I will. And I can try to make this a bit easier to read as well.